packages/agreement-process/src/services/agreementService.ts

```diff
@@ -503,3 +521,80 @@ export async function upgradeAgreementLogic({
     return [createEvent, ...docEvents];
   }
 }
+
+export async function cloneAgreementLogics({
+  agreementId,
+  authData,
+  agreementQuery,
+  tenantQuery,
+  eserviceQuery,
+  fileCopy,
+}: {
+  agreementId: string;
+  authData: AuthData;
+  agreementQuery: AgreementQuery;
+  tenantQuery: TenantQuery;
+  eserviceQuery: EserviceQuery;
+  fileCopy: (
+    container: string,
+    sourcePath: string,
+    destinationPath: string,
+    destinationFileName: string,
+    docName: string
+  ) => Promise<string>;
+}): Promise<Array<CreateEvent<AgreementEvent>>> {
+  const agreementToBeCloned = await agreementQuery.getAgreementById(
+    agreementId
+  );
+  assertAgreementExist(agreementId, agreementToBeCloned);
+  assertRequesterIsConsumer(
+    authData.organizationId,
+    agreementToBeCloned.data.consumerId
+  );
+
+  assertExpectedState(agreementId, agreementToBeCloned.data.state, [
+    agreementState.rejected,
+  ]);
+
+  const eservice = await eserviceQuery.getEServiceById(
+    agreementToBeCloned.data.eserviceId
+  );
+  assertEServiceExist(agreementToBeCloned.data.eserviceId, eservice);
+
+  const consumer = await tenantQuery.getTenantById(
+    agreementToBeCloned.data.consumerId
+  );
+  assertTenantExist(agreementToBeCloned.data.consumerId, consumer);
+
+  const descriptor = eservice.data.descriptors.find(
+    (d) => d.id === agreementToBeCloned.data.descriptorId
+  );
+  if (descriptor === undefined) {
+    throw descriptorNotFound(
+      eservice.data.id,
+      agreementToBeCloned.data.descriptorId
+    );
+  }
+
+  validateCertifiedAttributes(descriptor, consumer.data);
+
+  const newAgreement = await createAgreementLogic(
+    {
+      eserviceId: agreementToBeCloned.data.eserviceId,
+      descriptorId: agreementToBeCloned.data.descriptorId,
+    },
+    authData,
+    agreementQuery,
+    eserviceQuery,
+    tenantQuery
+  );
+
+  const docEvents = await createAndCopyDocumentsForClonedAgreement(
```

Comment:
I'm not able to find `createAndCopyDocumentsForClonedAgreement` implementation ...

Reply:
Sorry, I found it in previous development 😅